susemanager-utils/testing/automation/clean-objects.sh

```diff
@@ -1,25 +1,15 @@
 #/bin/bash -e
 echo "*************** CLEANING OBJECTS GENERATED INSIDE THE CONTAINER  ***************"
 # Exit if the file does not exist or if it is empty
-if [ ! -f /tmp/objects-init.txt -o -s /tmp/objects-init.txt ]; then
+if [ ! -f /tmp/objects-init.txt -o ! -s /tmp/objects-init.txt ]; then
   echo "No file with the initial list of objects"
   exit 0
 fi
-find /manager > /tmp/objects-end.txt
-while read OBJECTEND; do
-  FOUND=FALSE
-  while read OBJECTINIT; do
-    if [ "${OBJECTINIT}" == "${OBJECTEND}" ]; then
-      FOUND=TRUE
-      break
-    fi
-  done < /tmp/objects-init.txt
-  if [ "${FOUND}" == "FALSE" ]; then
-    echo "${OBJECTEND}"
-    if [ -d ${OBJECTEND} ]; then
-        rm -rf ${OBJECTEND}
-    else
-	rm -f ${OBJECTEND}
-    fi
+find /manager | sort > /tmp/objects-end.txt
+for LINE in $(diff /tmp/objects-init.txt /tmp/objects-end.txt|grep '> .*$'|sed -e 's/^> //'); do
```

Comment:
Basically: use simple diff output, grep those lines that identified added content, remove the prefix that identifies added content (`> `) and remove the files.